### PR TITLE
improved code structure 

### PR DESCRIPTION
**Pull request by dpshekhawat**
**Results of the PR:**

- More code readability.
- Set environment variables for -
> THRES -> for confidence threshold.
> TIMEINT -> for time interval of capturing frames.
> CAPTURE -> to stop capturing if set to False.
- [x]  Not burning any CPU while waiting around.
- [x] Handling Permission Error.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -2,68 +2,96 @@
...
+                (x, y),
+                (x + w, y + h),
+                (0, 200, 50),
+                thick,
+            )
+        """
+        Uncomment the below lines to see real-time 
+        feed of object detection.
+        """
+        #cv2.imshow("ObjectDetection", frame)
```
Comment (pkliczewski) [inline main.py]: Why do we need this commented code?

File: main.py [hunk 2]
```diff
@@ -2,68 +2,96 @@
...
+        #cv2.waitKey(1)
+
+        if int(round(time.time() - start_time, 2)) >= timeInterval:
+            filename = "{}.jpeg".format(datetime.now().strftime("%Y-%m-%d_%H-%M-%S"))
```
Comment (pkliczewski) [inline main.py:76]: I think we should use this if branch and get a frame and not only for writing.

File: main.py [hunk 3]
```diff
@@ -2,68 +2,96 @@
...
+# Path to store images
+folder = "images/"  
 
-if not os.path.exists(folder):
-    os.mkdir(folder)
+try:
+    if not os.path.exists(folder):
+        os.mkdir(folder)
+except PermissionError as pe:
+    print(pe)
```
Comment (pkliczewski) [inline main.py]: I think we should stop our script here. We won't be able to write any images if there is no permission or folder do not exists

File: main.py [hunk 4]
```diff
@@ -2,68 +2,96 @@
...
-        if capture != True:
-            break
+# cap.isOpened() to check cap object has started capturing the frame.
+while cap.isOpened() and capture:
+    # Get Frames
+    ret, frame = cap.read()
+
+    if ret:
+        # Object Detection
+        (class_ids, scores, bboxes) = model.detect(
```
Comment (pkliczewski) [inline main.py]: What do you think about having a function for object detection code.

File: main.py (hunk 1 above)
Comment (dpshekhawat) [reply main.py]: For developer preview of how the ML detection algo is working, but we can remove it.

Comment (pkliczewski) [reply main.py]: sure, we could keep it but we should expose env variable to and protect it with `if` checking for this variable. In general commented code case issues with readablility.

Comment (dpshekhawat) [reply main.py]: got it! will try to implement this.

File: main.py (hunk 3 above)
Comment (dpshekhawat) [reply main.py]: alright, adding it inside the loop so that it stops the script!

File: main.py (hunk 4 above)
Comment (dpshekhawat) [reply main.py]: working on it

File: main.py (hunk 1 above)
Comment (dpshekhawat) [reply main.py]: temporarily removed it.

File: main.py (hunk 2 above)
Comment (dpshekhawat) [reply main.py:76]: Modified to `time.sleep()` for saving CPU.

File: main.py [hunk 5]
```diff
@@ -2,68 +2,86 @@
...
+    # Get Frames
+    ret, frame = cap.read()
+
+    if ret:
+        # Error Handling
+        try:
+            if not os.path.exists(folder):
+                os.mkdir(folder)
+        except PermissionError as pe:
+            print("Cannot create folder "+str(pe))
```
Comment (pkliczewski) [inline main.py]: What do you think about using [f-string](https://docs.python.org/3/tutorial/inputoutput.html#tut-f-strings)

File: main.py [hunk 6]
```diff
@@ -2,68 +2,86 @@
...
-                cv2.imwrite(os.path.join(folder,filename), frame)
-                start_time = time.time()
-        if capture != True:
+
+# cap.isOpened() to check cap object has started capturing the frame.
+while cap.isOpened() and capture:
+    # Get Frames
+    ret, frame = cap.read()
+
+    if ret:
```
Comment (pkliczewski) [inline main.py]: read method returns `True` if object is read correctly ([reference](https://docs.opencv.org/4.x/dd/d43/tutorial_py_video_display.html)). We only attempt to detect object if we fail.

File: main.py [hunk 7]
```diff
@@ -2,68 +2,86 @@
...
+        # Error Handling
+        try:
+            if not os.path.exists(folder):
```
Comment (pkliczewski) [inline main.py]: I would move folder checking code outside of the loop. We do not want to do it every iteration.

**Review by pkliczewski — COMMENTED** [review]
It looks much better but adding 3 more comments

File: main.py (hunk 7 above)
Comment (dpshekhawat) [reply main.py]: Got it moved it out of loop and terminate the script if error occurs as suggested previously.

**Comment by pkliczewski** [comment]
Looks great, the change is ready to merge

**Pull request merged**